src/ring_theory/polynomial/chebyshev/dickson.lean

```diff
@@ -0,0 +1,187 @@
+/-
+Copyright (c) 2021 Julian Kuelshammer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Julian Kuelshammer
+-/
+
+import data.polynomial.eval
+import ring_theory.polynomial.chebyshev.defs
+
+/-!
+# Dickson polynomials
+
+The Dickson polynomials are two families of polynomials indexed by `ℕ`,
+with coefficients in a commutative ring `R` depending on an element `a∈R`.
+They are closely related to the Chebyshev polynomials in the case that `a=1`.
+When `a=0` they are just the family of monomials `x ^ n`.
+
+## Main declarations
+
+* `polynomial.dickson₁`: the Dickson polynomials of the first kind.
+* `polynomial.dickson₂`: the Dickson polynomials of the second kind.
+
+## References
+
+* [R. Lidl, G. L. Mullen and G. Turnwald, _Dickson polynomials_][MR1237403]
+
+## TODO
+
+* Move the definition of `lambdashev` from `chebyshev.defs` into this file
+  (or even remove it because it is a special case of the polynomials defined here?).
+* Show that `lambdashev` is equal to the characteristic polynomial of the adjacency matrix of a
+  type A Dynkin diagram.
+-/
+
+noncomputable theory
+
+namespace polynomial
+
+variables {R S : Type*} [comm_ring R] [comm_ring S]
+
+/-- `dickson'` is an auxiliary non-standard function that lets us unify some proofs for `dickson₁`
+and `dickson₂`.
+-/
+noncomputable def dickson' (p₀ : polynomial R) (a : R) : ℕ → polynomial R
+| 0       := p₀
+| 1       := X
+| (n + 2) := X * dickson' (n + 1) - (monomial 0 a) * dickson' n
```

Comment:
`monomial 0 a` is `C a`, right? That probably makes your life easier, because `C` is a ring hom.